### DISCUSSION TITLE
Call for ideas and proposals for the future of COPTR & COW

### DISCUSSION THREAD

**Discussion by anjackson** [digipres/digipres.github.io#53] [Announcements]
The Community Owned digital Preservation Tool Registry ([COPTR](https://coptr.digipres.org/)) is not sustainable in it's current form, and we need a plan for the future.

On a technical level, the shift to building on the Semantic MediaWiki stack has proven challenging to manage and maintain. The Open Preservation Foundation have done as much as possible to keep the service running, but the software dependencies and administrative complexity is too high. The system has also started to drown under a influx of spam content, which is difficult to keep out without causing other problems (see #36).

On a service level, it appears the tool list information itself is hard to use and to maintain. Most of the feedback I've had on COPTR has mentioned that the tool lists are often overwhelming. There can be so many options for some actions, and very little guidance on where to start.  This is compounded by other issues with the contents, like the fact that it is often outdated or incomplete, in the sense that it does _not_ include many widely-used tools or libraries.

In my opinion, the 'Tool Registry' paradigm itself encourages this, in that it encourages the sporadic accumulation of 'stub' entries that point to potentially interesting tools, but adding or refining these entries is not something that is easy to integrate into the work of digital preservation practitioners. Based on my experiences so far, there seem to be two other, more promising avenues:

- Connecting people with the communities that know about tools:
    - Asking for advice from active practitioners means the information is more up to date.
    - Most people like sharing their knowledge with like-minded peers.
- Sharing knowledge about _how tools are used, in context_:
    - Compared to flat lists of tools, information about how a tool is being used _for real_ is very valuable.
    - Quite a lot of people already like to spend time publishing their workflows in papers and blogs, and other forums like the DPC Workflow Webinar series.
    - This includes the [Community-Owned Workflows](https://coptr.digipres.org/index.php/Workflow:Community_Owned_Workflows) (COW), which also need a new home, as they are stored on COPTR.

So, I'm wondering if we can 'flatten' the COPTR tools list down enough that it becomes manageable (perhaps just turning it into an Awesome List for Digital Preservation Tools? Or Google Sheet?), and instead dedicate some effort into encouraging people to share workflows in useful ways? e.g. gather some of the existing Workflows and Guides together, grouped by different actions/goals, a bit like the [COPTR Tools Grid](https://coptr.digipres.org/index.php/Tools_Grid) does but at the Workflows level? (I'm planning to prototype something like this over the next few weeks!)

**Comment by ablwr** [comment]
Commenting to stay on this thread of thought :)

**Comment by ross-spencer** [comment]
This feels like it should be two or more separate conversations.

1) What to do with COPTR? (or COPTR 2.0)  
2) Are there other complementary approaches?  
3) What's needed by practitioners (or workflow developers) at different stages of their skills development?  
4) Miscellaneous, e.g., "connecting people" / “COW” / "webinars," etc.  

Regarding 2) COPTR and an awesome list / flattening of COPTR could and should, I believe, remain separate conversations. An awesome list for a community-driven "most used tools" sounds fine and is easily achievable and can be done as a complementary approach while bigger topics are tackled.

Concerning 1) (What to do with it), I don’t think this conversation should be shortcut. COPTR is an important resource and stands as its own registry. There's a lot to unpack; the slippery slope of "widely used tools" or libraries is a concern for anyone looking at the long tail of formats within their own institutional context. Also, as a tool developer, I have few places to represent my work (or novelties I find from others and contribute to COPTR) and am unlikely to make it onto a most-used list; a registry makes a lot of sense, even just as a flag in the ground highlighting the work. The idea of a registry lending itself to stubs does not seem like a bad thing, but it could definitely be enhanced in a COPTR “2.0”. Even if it's not in a Wiki, it could be easier to upvote, star review, or communicate about what's there, e.g., form-based submission for reviews/notes (think BBC Good Food Website for File Formats). The intentions of different tools could be clearer, e.g., nuancing function with regard to PUID.

As a whole, COPTR (1.0) seems to me to be about a discussion concerning marketing, support, identifying ownership and responsibility, opening maintenance, community building, and awareness building. To begin, I didn't really know it was being very actively maintained, and I couldn't point you to an individual to talk to about it until I asked about the spam situation—I thought its [relaunch in 2021](https://openpreservation.org/news/an-invitation-to-the-coptr-relaunch/) might signal a committed phase of maintenance, contribution, and support from the DPC. The spam situation alone doesn't seem to be a dealbreaker; that could be any Wiki on the internet; I feel like that conversation shouldn’t be exhausted.

That being said, it doesn't mean COPTR can’t be re-architected. Even if it's not, the moment feels instructional to the community, especially since the relaunch, as to what worked, what didn't work, what could be better, and so on. I don't know; losing stubs to 600+ tools seems like an unnecessary narrowing of the landscape. Sunsetting or changing the approach might also warrant a conversation about migrating web links (or maintaining redirects for a period) so that individuals who point to COPTR don’t end up with broken links. Maybe it can simply be put into read-only mode? And could exporting to Wikidata/WikiDP be considered?

(It feels like a big subject, and the more I think about it, I wonder if it’s a conversation that should be hosted as an event, e.g., via the OPF? For greater awareness. Also, will you be sharing this discussion via your mailing lists?)

(Additionally, it’s not clear to me how much PAR should be invoked in this conversation, but there might be some synergy.)

On 4) Connecting people / COW and so on, and sharing tools is beyond what I can helpfully discuss; obviously, it shouldn't be restricted to a membership organization. And as a proponent of linked open data, context looks very different when backed up by URLs that can be dereferenced and provide more information, so something COPTR-like is always going to support that, and combining community-described workflows with pointers to tools can become a self-reinforcing approach.

On 3) (practitioner needs at different stages), yes, this just feels like something different from COPTR. A different conversation, probably not for me in any current capacity, and more for your own outreach, members, etc. It's not impossible to reorganize COPTR or create new entry points and pathways into the data. It also feels like it could be a different, make-your-own-adventure-like tool. I can definitely see a low-barrier entry point benefiting a COPTR-like service; but there’s also an advanced view as a workflow builder / architect. where you do want to see the landscape that exists to know what you can and cannot put together or what gaps exist.

**Comment by ross-spencer** [comment]
@anjackson @carlwilson this is probably the best way I have to ping either of you at the same time right now, but COPTR has been on maintenance mode for over a week now so no new entries can be added. Can it be switched back to normal mode?

NB. I don't think there has been an announcement? but if it's a long-term pause while the technical issues are resolved, and this discussion above is had, then it's probably a good idea to make one. (and of course lmk if there has been something announced somewhere, I just haven't seen anything on different networks)

**Reply by anjackson** [reply]
Fair point Ross. We've struggled to coordinate on this, due to other pressures. I'll try to send something out onto the JISC mailing list.

_EDIT_ I've queued an email to go out on Monday morning.

**Comment by anjackson** [comment]
As outlined here: https://www.jiscmail.ac.uk/cgi-bin/wa-jisc.exe?A2=DIGI-PRES-REGISTRIES;2470e892.2502 I'm dedicating the March 6th Preservation Registries SIG meetings to talking about this. All are welcome!

We (the DPC) have had problems in the past with making video-conference links fully public (i.e. [Zoombombing](https://en.wikipedia.org/wiki/Zoombombing)), so you currently have to [sign up for that mailing list](https://www.jiscmail.ac.uk/cgi-bin/webadmin?A0=DIGI-PRES-REGISTRIES) to access the details. Sorry!

**Comment by karinbredenberg** [comment]
I would say that for PREMIS and METS I see COPTR as a valuable resource to point to and we do point to it since maintain one place with tools is (should) be easier than having multiple places. There is now a couple of years ago since we in both groups decided that COPTR was the place for the tools instead of our webpages and we added the info that was in our lists but not in COPTR to COPTR. Can confirm what you write @anjackson what we saw when we did have our own list of tools on our webpages was that we also had real difficulties to keeping things accurate, contact persons leaving their positions and so on. So for us instead pointing to COPTR and say please add you tool here and add in the markers for saying that the tool is using METS or PREMIS turned out in the best way. Gives more responsibility but at the same time makes it easier to do updates for the tool creator which as you write also is tricky to get going. 

I've said it before, and do it again here, thanks for the help with those markers!

So I don't want to see COPTR vanish since where should we point people who want to find tools if there is no COPTR but are at the same time not the person to say what should be the next phase.

**Reply by anjackson** [reply]
Thanks @karinbredenberg - COPTR isn't going to disappear! We're just not sure what form it should take. But information like this will help us understand what the real user needs are, so we can try to find a good fit! Thank you.
